### PR TITLE
Add remove item

### PR DESCRIPTION
**Pull request by thecodechaser**
This branch includes the following features:
Added functionality to add items to the to-do list.
Added functionality to remove items to the to-do list.
Users can check the tasks that are completed.
Users can delete all the completed tasks at once.
Users can update the values of tasks.
Delete option shows up on the screen when user mark task as completed.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello @thecodechaser 

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._


Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/utils/interaction.js [hunk 1]
```diff
@@ -1,25 +1,42 @@
-import { items, displayTask } from './structure.js';
+/* eslint-disable import/no-cycle */
```
Comment (mikenath223) [inline src/utils/interaction.js]: - [ ] Kindly remove this eslint disable line its best to fix the issue in the code.

File: src/utils/interaction.js [hunk 2]
```diff
@@ -1,25 +1,42 @@
...
+    if (item.hasAttribute('checked')) {
+      item.nextSibling.style.textDecoration = 'line-through';
+    } else {
+      item.nextSibling.style.textDecoration = 'none';
+    }
+  });
+}
 
-function userInteraction() {
+function userInteraction(listInput) {
```
Comment (mikenath223) [inline src/utils/interaction.js]: - [ ] Please change this and all other regular functions into arrow functions, following the Javascript best practices guide.

File: src/utils/structure.js [hunk 3]
```diff
@@ -1,50 +1,102 @@
...
-    index: 3,
-  },
-  {
-    description: 'Dinner plan with family at hotel',
-    completed: false,
-    index: 4,
-  },
-];
+/* eslint-disable import/no-cycle */
+/* eslint-disable no-use-before-define */
```
Comment (mikenath223) [inline src/utils/structure.js]: - [ ] Kindly remove this eslint disable line its best to fix the issue in the code.

File: src/utils/structure.js [hunk 4]
```diff
@@ -1,50 +1,128 @@
...
+        items.splice(i, 1);
+      }
+    }
+    for (let i = 0; i < items.length; i += 1) {
+      items[i].index = i + 1;
+    }
+    localStorage.setItem('itemsLocal', JSON.stringify(items));
+    render();
+  });
+};
```
Comment (D4nielJ) [inline src/utils/structure.js]: - [ ] It seems like the "Clear all completed" functionality it's not working properly. When I click on the button the tasks that are marked as completed remain in the list.

> A function for the "Clear all completed" button using the `Array.filter()` method is implemented.

todo-list

**Review by D4nielJ — CHANGES_REQUESTED** [review]
## Hello @thecodechaser, 

Great work on making the changes requested by the previous reviewer(s) 👏

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Happy coding and stay awesome! 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikenath223 — APPROVED** [review]
Hello there  

Your project is complete!  There is nothing else to say other than... it's time to merge it 🛳 
Congratulations! 🎉 

Happy coding. Cheers! 🚀🚀

If you feel something isn't clear enough you can leave a question or comment in the PR thread.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**